### PR TITLE
[DOC] add missing dataset document

### LINKED ISSUE #831: ATIS/SNIPS datasets and GLUE datasets don't appear in the website API doc
http://gluon-nlp.mxnet.io/api/modules/data.html

does not show the details of ATISDataset/SNIPSDataset and GlueCoLA, GlueSST2, GlueSTSB, GlueQQP, GlueRTE, GlueMNLI, GlueQNLI, GlueWNLI

### PR DESCRIPTION
**Pull request by szha**
## Description ##
add missing dataset document. resolves #831 

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [x] add missing dataset document

### CODE REVIEW

**Comment by mli** [comment]
Found link check problems in job PR-832/1:
(line    7) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    3) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    7) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line   19) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line   21) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line  442) broken    https://www.aclweb.org/anthology/P02-1040.pdf)[1 - 404 Client Error: Not Found for url: https://www.aclweb.org/anthology/P02-1040.pdf)%5B1
(line   92) broken    https://nlp.stanford.edu/pubs/glove.pdf)[2 - 404 Client Error: NOT FOUND for url: https://nlp.stanford.edu/pubs/glove.pdf)%5B2
(line  208) broken    https://www.bioinf.jku.at/publications/older/2604.pdf)[3 - 404 Client Error: Not Found for url: https://www.bioinf.jku.at/publications/older/2604.pdf)%5B3

**Comment by mli** [comment]
Job PR-832/1 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-832/1/index.html

File: docs/api/modules/data.rst [hunk 1]
```diff
@@ -140,6 +140,24 @@ Intent Classification and Slot Labeling
     ATISDataset
```
Comment (eric-haibin-lin) [inline docs/api/modules/data.rst]: Also move SQuAD out of Machine Translation Datasets ?

**Comment by mli** [comment]
Found link check problems in job PR-832/2:
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    3) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    7) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    7) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line    4) broken    https://gluebenchmark.com/tasks - 404 Client Error: Not Found for url: https://gluebenchmark.com/tasks
(line   19) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line    5) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line   21) broken    https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/ - 404 Client Error: Not Found for url: https://www.salesforce.com/products/einstein/ai-research/the-wikitext-dependency-language-modeling-dataset/
(line  442) broken    https://www.aclweb.org/anthology/P02-1040.pdf)[1 - 404 Client Error: Not Found for url: https://www.aclweb.org/anthology/P02-1040.pdf)%5B1
(line   92) broken    https://nlp.stanford.edu/pubs/glove.pdf)[2 - 404 Client Error: NOT FOUND for url: https://nlp.stanford.edu/pubs/glove.pdf)%5B2
(line  208) broken    https://www.bioinf.jku.at/publications/older/2604.pdf)[3 - 404 Client Error: Not Found for url: https://www.bioinf.jku.at/publications/older/2604.pdf)%5B3

**Comment by mli** [comment]
Job PR-832/2 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-832/2/index.html

**Pull request merged**